core/src/com/unciv/models/ruleset/unit/BaseUnit.kt

```diff
@@ -39,14 +39,18 @@ class BaseUnit : RulesetObject(), INonPerpetualConstruction {
     var unitType: String = ""
 
     val type by lazy { ruleset.unitTypes[unitType]!! }
-    override var requiredTech: String? = null
+    // Warning: future development should not increase the role of requiredResource, and should reduce it when possible.
+    // The Unique "Consumes [1] [$requiredResource]" accomplishes the same task.
+    // https://yairm210.github.io/Unciv/Developers/Translations%2C-mods%2C-and-modding-freedom-in-Open-Source#filters
     var requiredResource: String? = null
 
     override fun getUniqueTarget() = UniqueTarget.Unit
 
     var replacementTextForUniques = ""
     var promotions = HashSet<String>()
-    var obsoleteTech: String? = null
+    // Warning: future development should not increase the role of obsoleteTech, and should reduce it when possible.
+    // The Unique "Only available before discovering [$obsoleteTech]" accomplishes the same task.
```

Comment:
Not sure this is true. There is mechanic baggage regarding obsoleting objects, at least in the form of production switching to what units upgrade into

This assumption is unnecessary. `UniqueType.ObsoleteWith` covers obsoleting functionality. If you're migrating fields to uniques, that UniqueType should be included in the discussion